src/peer/mod.rs

```diff
@@ -447,13 +450,20 @@ impl PeerConnection {
         {
             let id = self.id;
             let weak_sender = Rc::downgrade(&self.peer_events_sender);
+            let connections = Rc::downgrade(&self.connections);
             self.peer.on_connection_state_change(Some(
                 move |peer_connection_state| {
                     if let Some(sender) = weak_sender.upgrade() {
+                        let member_ids = connections
+                            .upgrade()
+                            .map(|connections| connections.member_ids())
+                            .unwrap_or_default();
+
                         Self::on_connection_state_changed(
                             id,
                             &sender,
                             peer_connection_state,
+                            member_ids,
```

Comment:
This `member_id` will contain all the members in the current room. E.g. if there is 5 peers and 5 corresponding members, if one peer goes to State::Failed, then this state change will update all ther currently available connections.

This could make sense in SFU mode, since all connections are backed by just two peers, but this is definitely wrong for the P2P scenarion, when there is a single peer for each connection/member.